### PR TITLE
[PF-2929] Add dataproc cluster operations

### PR DESCRIPTION
**Pull request by rogerwangcs**
Adds endpoints for start, stop, status, and jupyter component proxy url.

### CODE REVIEW

File: service/src/test/java/bio/terra/axonserver/utils/dataproc/GoogleDataprocClusterTest.java [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    var mockStart = mock(DataprocCow.Clusters.Start.class, finalMockSettings);
+
+    // Wire the request mock to return the fake operation
+    Mockito.when(mockStart.execute()).thenReturn(mockOperation);
+
+    // Wire the mock Clusters to return the fake request
+    Mockito.when(mockClusters.start(clusterName)).thenReturn(mockStart);
+
+    // Call the method
+    cluster.start(false);
```
Comment (yuhuyoyo) [inline service/src/test/java/bio/terra/axonserver/utils/dataproc/GoogleDataprocClusterTest.java]: /*fieldName=*/ false

File: service/src/test/java/bio/terra/axonserver/utils/dataproc/GoogleDataprocClusterTest.java [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+    ArgumentCaptor<Operation> operationArgumentCaptor = ArgumentCaptor.forClass(Operation.class);
+    ArgumentCaptor<String> errorMessageArgumentCaptor = ArgumentCaptor.forClass(String.class);
+    Mockito.verify(cluster)
+        .pollForSuccess(operationArgumentCaptor.capture(), errorMessageArgumentCaptor.capture());
+    assertEquals(mockOperation, operationArgumentCaptor.getValue());
+    assertTrue(errorMessageArgumentCaptor.getValue().contains(operation));
+  }
+
+  @Test
+  public void start() throws IOException {
```
Comment (yuhuyoyo) [inline service/src/test/java/bio/terra/axonserver/utils/dataproc/GoogleDataprocClusterTest.java]: start_noWait_doNotPull

File: service/src/test/java/bio/terra/axonserver/utils/dataproc/GoogleDataprocClusterTest.java [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+
+    // Call the method
+    cluster.start(false);
+
+    // Make sure pollForSuccess was not called.
+    Mockito.verify(cluster, never()).pollForSuccess(any(), any());
+  }
+
+  @Test
+  public void start_wait() throws IOException {
```
Comment (yuhuyoyo) [inline service/src/test/java/bio/terra/axonserver/utils/dataproc/GoogleDataprocClusterTest.java]: start_wait_pull

File: service/src/test/java/bio/terra/axonserver/utils/dataproc/GoogleDataprocClusterTest.java (hunk 2 above)
Comment (yuhuyoyo) [reply service/src/test/java/bio/terra/axonserver/utils/dataproc/GoogleDataprocClusterTest.java]: also separate your code into three blocks:
setup
test
assert
thanks :)

File: service/src/test/java/bio/terra/axonserver/utils/dataproc/GoogleDataprocClusterTest.java [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+    cluster.start(true);
+
+    // Verify that pollForSuccess was called with the right operation error message.
+    verifyPollForSuccess("start");
+  }
+
+  @Test
+  public void start_throw() throws IOException {
+    Mockito.when(mockClusters.start(clusterName)).thenThrow(new IOException());
+    try {
```
Comment (yuhuyoyo) [inline service/src/test/java/bio/terra/axonserver/utils/dataproc/GoogleDataprocClusterTest.java:112]: can you use assertThrow

File: service/src/test/java/bio/terra/axonserver/utils/dataproc/GoogleDataprocClusterTest.java [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+      cluster.start(true);
+      fail("Expected InternalServerErrorException");
+    } catch (InternalServerErrorException e) {
+      assertTrue(e.getMessage().contains("start"));
+    }
+  }
+
+  @Test
+  public void stop() throws IOException {
```
Comment (yuhuyoyo) [inline service/src/test/java/bio/terra/axonserver/utils/dataproc/GoogleDataprocClusterTest.java]: stop_noWait_notPull

File: service/src/test/java/bio/terra/axonserver/utils/dataproc/GoogleDataprocClusterTest.java [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+public class GoogleDataprocClusterTest {
+  private final ClusterName clusterName =
+      ClusterName.builder()
+          .projectId("fakeproject")
+          .region("fakeregion")
+          .name("fakeclustername")
+          .build();
+
+  /**
+   * Settings required to mock final classes/methods, required for several GCP SDK objects. Requires
```
Comment (yuhuyoyo) [inline service/src/test/java/bio/terra/axonserver/utils/dataproc/GoogleDataprocClusterTest.java:42]: never seen this. is this for dataprocCow.cluster and Operation?

**Review by yuhuyoyo — APPROVED** [review]

File: service/src/test/java/bio/terra/axonserver/utils/dataproc/GoogleDataprocClusterTest.java (hunk 6 above)
Comment (rogerwangcs) [reply service/src/test/java/bio/terra/axonserver/utils/dataproc/GoogleDataprocClusterTest.java:42]: Following a pattern John C dith sagemaker+vertex ai nb operations

File: service/src/test/java/bio/terra/axonserver/utils/dataproc/GoogleDataprocClusterTest.java (hunk 2 above)
Comment (rogerwangcs) [reply service/src/test/java/bio/terra/axonserver/utils/dataproc/GoogleDataprocClusterTest.java]: this was a bit of test code dupe but will do

**Review by harrismendell — APPROVED** [review]

**Pull request merged**